chapters/statemachines.tex

```diff
@@ -113,20 +113,29 @@ \section{Transitions}\doublelabel{transitions}
 
 It is an error if the instance is not a state of a state machine.\\ \hline
 \textbf{ticksInState()} & Returns the number of ticks of the clock of
-the state machine since a transition was made to the currently active
-state. This function can only be used in transition conditions of state
-machines not present in states of hierarchical state machines.
-{[}\emph{For example if there is a transition with immediate=false from
-state A1 to A2 and the condition is ticksInState() $\ge 5$, and A1 became
-active at 10ms, and the clock period is 1ms, then A1 will be active at
-10ms, 11ms, 12ms, 13ms, 14ms, and will be not active at 15 ms.}{]}\\ \hline
+the state machine since the active state or the enclosing state (in a hierarchical state machine) was last entered.
+This function can only be used in state machines.\\ \hline
```

Comment:
I assume the intention to change this has been discussed somewhere, but unfortunately I don't know or remember there.

I agree to relax the unnecessary restriction imposed on operators `ticksInState` and `timeInState`. However, is there a reasoning why the operators should return the ticks (or time) of the _enclosing_ state rather than the _active_ state if the state is placed in a hierarchical state machines?

Reply:
I have tried to improve the formulation; it should since state became active - either due to local transitions or enclosing ones. See #2407 for more info.